herd/AArch64Sem.ml

```diff
@@ -391,6 +409,65 @@ module Make
 
       let write_reg_scalable = write_reg_scalable_sz MachSize.S128
 
+      let za_getoffset tile slice idx esize =
+        let dim = scalable_nbits / esize in
+        let tile = V.op Op.Mul (V.intToV tile) (V.intToV (dim*dim)) in
+        let tiles = V.intToV (esize/8) in
+         M.op Op.Mul slice tiles >>= fun v ->
+          M.op Op.Add tile v  >>= fun v ->
+            M.op Op.Add v idx >>= fun v ->
+              M.op Op.Mul v (V.intToV esize)
```

Comment:
Here is my patch, however beware,  the output of tests V12, V15 and V16 have changed.
```suggestion
      let za_getoffset tile slice idx esize =
        (* ZA offset, in bits, see ARM ARM B.1.4.10 "ZA tile access" *)
      let za_getoffset tile slice idx esize =
        let esize_to_shift = function
          | 8 -> 0
          | 16 -> 1
          | 32 -> 2
          | 64 -> 3
          | 128 -> 4
          | _ -> assert false in
        let shift = esize_to_shift esize in
        let mk_shift k = M.op1 (Op.LeftShift k) in
          begin
            begin
              mk_shift shift slice >>=
              M.add (V.intToV tile) >>=
              M.op Op.Mul (V.intToV scalable_nbits)
            end >>|
            mk_shift (shift+3) idx
          end >>= fun (base,idx) -> M.add base idx

```

Reply:
Nipicking: one can use `M.op1 (Op.AddK tile)` in place of `M.add (V.intToV tile)`.

Reply:
I agree. Both my **qemu** give these results.